stabbity.lic

```diff
@@ -0,0 +1,253 @@
+    @current_weapon = ''
+    @arena_trap = nil
+    
+    cleanup if cleanup_mode?
+    use_weapon
+    exit if equip_mode?
+    hide?
+    combat_loop
+  end
+
+  def arena_mode?
+    @mode == 'arena'
+  end
+
+  def single_mode?
+    @mode == 'single'
+  end
+
+  def equip_mode?
+    @mode == 'equip'
+  end
+
+  def cleanup_mode?
+    @mode == 'cleanup'
+  end
+
+  def cleanup
+    echo "*** Cleaning up ***" if @debug
+    @equipmanager.stow_weapon
+    exit
+  end
+
+  def combat_loop
+    loop do
+      npcs = get_npcs
+      if npcs.count > 0
+        @target = npcs.first
+        @npc_is_alive = true
+        echo "*** Target is #{@target} ***" if @debug
+        watch_target if arena_mode?
+        if @alternate_weapon_npcs.include?(@target) || @thrown_weapon_npcs.include?(@target)
+          dodge_arena_trap if arena_mode?
+        end
+        case bput("advance #{@target}", 'You are already at melee', 'is already quite dead', 'You begin to', 'You spin around')
+        when 'is already quite dead'
+          pause 2
+        end
+        using_right_weapon?
+        if @thrown_weapon_npcs.include? @target
+          echo "*** Target #{@target} is in thrown_weapon_npcs ***"
+          kill_thrown
+          waitrt?
+          loot_mob unless @noloot
+          use_weapon 'preferred'
+          hide?
+        else
+          kill_stabbity
+          waitrt?
+          loot_mob unless @noloot
+          use_weapon 'preferred'
+          hide?
+        end
+      else
+        echo "*** No targets ***" if @debug
+        exit if single_mode?
+        pause 1
+      end
+    end
+  end
+
+  def kill_stabbity
+    while @npc_is_alive
+      fix_standing
+      case bput('backstab', 'You must be hidden to blindside', 
+                            'It would help if you were closer',
+                            'You\'ll need to stand up first',
+                            'flying out of reach',
+                            'to stop flying before',
+                            'Roundtime',
+                            /\[You\'re /,
+                            'There is nothing else')
+      when 'There is nothing else'
+        @npc_is_alive = false
+        return
+      when 'You must be hidden to blindside'
+        hide?
+        waitrt?
+      when 'It would help if you were closer'
+        fput('advance')
+        pause 3
+      when 'You\'ll need to stand up first'
+        fix_standing
+        hide?
+      when 'flying out of reach', 'to stop flying before'
+        echo "*** Switch to thrown -- add this MOB to your use_thrown_on list ***" if @debug
+        use_weapon 'thrown'
+        kill_thrown
+        return
+      when 'Roundtime', /\[You\'re /
+        return if npc_dead?
+        waitrt?
+        dodge_arena_trap if arena_mode?
+        hide?
+      end
+    end
+  end
+
+  def kill_thrown
+    action = @settings['stabbity']['thrown_action'].nil? ? 'throw' : @settings['stabbity']['thrown_action']
```

Comment:
Should the default throw action be to 'lob' to avoid lodging?

Reply:
Yes. Next release will use EquipmentManager to read the lodging/bound parameters and copy `combat-trainer`'s `thrown_attack_verb` and `thrown_retrieve_verb` methods.